nengo_dl/converter.py

```diff
@@ -1248,31 +1220,41 @@ def convert(self, node_id, dimensions):
             # conv layer biases are per-output-channel, rather than per-output-element,
             # so we need to set up a nengo connection structure that will have one
             # bias parameter shared across all the spatial dimensions
-            if self.layer.data_format == "channels_first":
-                spatial_size = np.prod(self.output_shape(node_id)[1:])
-                bias_node = nengo.Node(np.ones(spatial_size), label="conv_bias")
-                offset = 0
-                for i in range(self.output_shape(node_id)[0]):
-                    nengo.Connection(
-                        bias_node,
-                        output[offset : offset + spatial_size],
-                        transform=biases[i],
-                        synapse=None,
-                    )
-                    offset += spatial_size
-            else:
-                spatial_size = np.prod(self.output_shape(node_id)[:-1])
-                bias_node = nengo.Node(np.ones(spatial_size), label="conv_bias")
-                idxs = np.arange(np.prod(self.output_shape(node_id))).reshape(
-                    (-1, self.output_shape(node_id)[-1])
+
+            # add trainable bias weights
+            bias_node = nengo.Node([1])
+            bias_relay = nengo.Node(size_in=len(biases))
+            nengo.Connection(
+                bias_node, bias_relay, transform=biases[:, None], synapse=None
+            )
+
+            # use a non-trainable sparse transform to broadcast biases along all
+            # non-channel dimensions
+            broadcast_indices = []
+            idxs = np.arange(np.prod(self.output_shape(node_id))).reshape(
+                self.output_shape(node_id)
+            )
+            slices = [slice(None) for _ in range(len(self.output_shape(node_id)))]
+            n_spatial = np.prod(
+                self.output_shape(node_id)[:-1]
+                if self.layer.data_format == "channels_last"
+                else self.output_shape(node_id)[1:]
+            )
+            axis = -1 if self.layer.data_format == "channels_last" else 0
+            for i in range(self.output_shape(node_id)[axis]):
+                slices[axis] = i
+                broadcast_indices.extend(
+                    tuple(zip(np.ravel(idxs[tuple(slices)]), [i] * n_spatial))
                 )
-                for i in range(self.output_shape(node_id)[-1]):
-                    nengo.Connection(
-                        bias_node,
-                        output[idxs[:, i]],
-                        transform=biases[i],
-                        synapse=None,
-                    )
+            conn = nengo.Connection(
+                bias_relay,
+                output,
+                transform=nengo.Sparse(
+                    (output.size_in, bias_relay.size_out), indices=broadcast_indices
```

Comment:
I would explicitly set `init=1` here, just to make it clear what the sparse values are.

Reply:
I assumed that it was 1, personally. If we thought it would be ambiguous we should have made `init` a required argument in `nengo.Sparse`, but since that didn't come up when working on that, I feel like it's what we would naturally assume.